scripts/manage_subscription.py

```diff
@@ -0,0 +1,174 @@
+import click
+from ape import Contract
+from ape.cli import account_option, ConnectedProviderCommand
+
+from deployment import registry
+from deployment.options import (
+    subscription_contract_option,
+    domain_option,
+    access_controller_option,
+    ritual_id_option,
+    encryptor_slots_option,
+    encryptors_option,
+)
+from deployment.params import Transactor
+from deployment.utils import check_plugins
+
+
+def _erc20_approve(
+        amount: int,
+        erc20: Contract,
+        receiver: Contract,
+        transactor: Transactor
+) -> None:
+    """Approve an ERC20 transfer."""
+    click.echo(
+        f"Approving transfer of {amount} {erc20.contract_type.name} "
+        f"to {receiver.contract_type.name}."
+    )
+    transactor.transact(
+        erc20.approve,
+        receiver.address,
+        amount
+    )
+
+
+def _calculate_slot_fees(
+        subscription_contract: Contract,
+        slots: int
+) -> int:
+    """Calculate the fees for a given number of encryptor slots."""
+    duration = subscription_contract.subscriptionPeriodDuration()
+    encryptor_fees = subscription_contract.encryptorFees(slots, duration)
+    total_fees = encryptor_fees
+    return total_fees
+
+
+@click.group()
+def cli():
+    """Subscription Management CLI"""
+
+
+@cli.command(cls=ConnectedProviderCommand)
+@account_option()
+@domain_option
+@subscription_contract_option
+@encryptor_slots_option
+@click.option(
+    "--period",
+    default=0,
+    help="Subscription billing period number to pay for.",
+)
+def pay_subscription(account, domain, subscription_contract, encryptor_slots, period):
```

Comment:
If you wanted these commands to not use dashes in the CLI you could use the name option in the command() directive. For example, for this function you could do:
```python
@cli.command(name=paySubscription, cls=ConnectedProviderCommand)
...
```
Then the command would not use any dashes.

You can do the same for the `click.group()` i.e. `click.group(name="manageSubscription") for the `manage_subscription` CLI command group which derives its name from the file by default.

Up to you though - just giving options.

Reply:
I gave this a try for the `click.group` but I did not have any success 🤷🏻 

```python
@click.group(name="manage-subscription")
def cli():
    """Subscription Management CLI"""
```

```python
(nucypher-contracts) k@Kierans-MacBook-Pro nucypher-contracts % ape run manage-subscription --help
Usage: ape run [OPTIONS] COMMAND [ARGS]...
Try 'ape run -h' for help.

Error: No such command 'manage-subscription'. Did you mean manage_subscription?
```

Reply:
Interesting. I guess the cli group name doesn't matter to ape when doing `ape run ...`.

Reply:
The "name" property for click.command should still work though (if desired).

Reply:
For sure, it works.  Personally I prefer the default naming behavior for CLI commands.